hie-base/Haskell/Ide/Engine/SemanticTypes.hs

```diff
@@ -62,6 +62,18 @@ data ModuleList = ModuleList {
   } deriving (Show,Read,Eq,Ord,Generic)
 
 -- ---------------------------------------------------------------------
+
+-- | GHC AST
+data AST = AST {
+    astModule :: T.Text
+  , astParsed      :: Value
+  , astRenamed     :: Value
```

Comment:
ok, and I looked more closely, we are receiving it as JSON, so presumably
the absence is already encoded.

On Wed, Jan 13, 2016 at 9:49 AM, Moritz Kiefer <notifications@github.com>
wrote:

> In hie-base/Haskell/Ide/Engine/SemanticTypes.hs
> <https://github.com/haskell/haskell-ide-engine/pull/170#discussion_r49559446>
> :
>
> > @@ -62,6 +62,18 @@ data ModuleList = ModuleList {
> >    } deriving (Show,Read,Eq,Ord,Generic)
> >
> >  -- ---------------------------------------------------------------------
> > +
> > +-- | GHC AST
> > +data AST = AST {
> > +    astModule :: T.Text
> > +  , astParsed      :: Value
> > +  , astRenamed     :: Value
>
> Maybe Values are annoying since both Nothing and Just Null get serialized
> to Null. Also the Null in the Value makes the Nothing redundant at least
> as long as we don’t need two different null cases (I don’t know if that’s
> the case here).
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/haskell/haskell-ide-engine/pull/170/files#r49559446>.
>